### PR TITLE
Use the OS' native font by default

### LINKED ISSUE #3970: Support for the system UI font
The font selection for styles only permits a fixed set regardless of the user's platform preferences. Offering the selection of the system UI has the benefit of adhering to the user selection without relying on rather expensive web fonts.

### PR DESCRIPTION
**Pull request by dtdesign**
Resolves #3970

/cc @SoftCreatR (#4017)

### CODE REVIEW

**Comment by SoftCreatR** [comment]
That's a pretty unusual stack, but better than before 😄 

Maybe use GitHub's?

Default: `-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji`
Monospace: `SFMono-Regular,Consolas,Liberation Mono,Menlo,monospace`

**Comment by dtdesign** [comment]
The GitHub stack does not provide native fonts for Android and Linux, they would fall back to "Helvetica" or "Arial". I don't understand GitHub's motivation of excluding those, especially Android has _quite some_ market share?

**Comment by SoftCreatR** [comment]
Yeah, GH is still not mobile optimized :/

**Comment by dtdesign** [comment]
For the record, the implementation is designed to allow future modifications of the font stack to address changes in case new fonts emerge. I do not think that GitHub's font stack is better, the font stack proposed in this PR covers their font plus a few more for Android/Linux, but please feel free to suggest changes if you think I missed something here.

**Comment by SoftCreatR** [comment]
It's all good, I like the proposed changes.

File: wcfsetup/install/files/lib/system/style/StyleCompiler.class.php [hunk 1]
```diff
@@ -582,14 +582,26 @@ protected function compileStylesheet(string $scss, array $variables): string
...
             $variables['wcfFontFamilyGoogle'] = '"' . $variables['wcfFontFamilyGoogle'] . '"';
 
             $variables['wcfFontFamily'] = $variables['wcfFontFamilyGoogle'] . ', ' . $variables['wcfFontFamily'];
         }
 
+        // Define the font family set for the OS default fonts. This needs to be happen statically to
+        // allow modifications in the future in case of changes.
+        $variables['wcfFontFamilyMonospace'] = 'ui-monospace, Menlo, Monaco, "Cascadia Mono",
+            "Segoe UI Mono", "Roboto Mono", "Oxygen Mono", "Ubuntu Monospace", "Source Code Pro",
+            "Fira Mono", "Droid Sans Mono", "Courier New", monospace';
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/system/style/StyleCompiler.class.php:597]: Maybe move this up into `private const` class constants. That way one does not have to search the whole file for these strings.

File: wcfsetup/install/files/lib/system/style/StyleCompiler.class.php [hunk 2]
```diff
@@ -582,14 +582,26 @@ protected function compileStylesheet(string $scss, array $variables): string
...
+
+        if ($variables['wcfFontFamily'] === 'system') {
```
Comment (TimWolla) [inline wcfsetup/install/files/lib/system/style/StyleCompiler.class.php:599]: Maybe use a `public const` for the string `system` to allow for re-use in StyleAddForm.

**Review by TimWolla — APPROVED** [review]
Overall this LGTM.

**Review by TimWolla — APPROVED** [review]

**Pull request merged**